kebechet/source_management.py

```diff
@@ -100,90 +114,65 @@ def close_issue_if_exists(self, title: str, comment: str = None):
-                'description': body,
-                'source_branch': branch_name,
-                'target_branch': 'master',
-                'allow_collaboration': True
-            }
+    def _gitlab_assign(self, issue: Issue, assignees: typing.List[str]) -> None:
+        """Assign the given users to a particular issue. Gitlab assignee id's are different from username."""
+        assignees = [int(assignee) for assignee in assignees]
+        data = {"assignees": assignees}
+        response = requests.Session().put(
+            f"{BASE_URL['gitlab']}/repos/{self.slug}/issues/{issue._raw_issue.iid}/assignees",
+            headers={f"Authorization": f"token {self.token}"},
+            json=data,
         )
-        try:
-            response.raise_for_status()
-        except Exception as exc:
-            raise RuntimeError(f"Failed to create a pull request: {response.text}") from exc
 
-        mr_number = response.json()['iid']
-        _LOGGER.info(f"Newly created pull request #{mr_number} available at {response.json()['web_url']}")
-        return GitLabMergeRequest.from_data(
-            response.json(), token=GitLabPrivateToken(self.token), repository=self.slug, number=mr_number
-        )
+        response.raise_for_status()
 
     def assign(self, issue: Issue, assignees: typing.List[str]) -> None:
         """Assign users (by their accounts) to the given issue."""
+        # Replace with OGR methods, when implemented in OGR.
         if self.service_type == ServiceType.GITHUB:
-            users = (GitHubUser(GitHubToken(self.token), username) for username in assignees)
+            self._github_assign(issue, assignees)
         elif self.service_type == ServiceType.GITLAB:
-            users = (GitLabUser(GitLabPrivateToken(self.token), username) for username in assignees)
+            self._gitlab_assign(issue, assignees)
         else:
             raise NotImplementedError
 
-        issue.assign(*users)
-
-    def open_merge_request(self, commit_msg: str, branch_name: str, body: str, labels: list) -> MergeRequest:
+    def open_merge_request(
+        self, commit_msg: str, branch_name: str, body: str, labels: list
+    ) -> PullRequest:
         """Open a merge request for the given branch."""
-        if self.service_type == ServiceType.GITHUB:
-            merge_request = self._github_open_merge_request(commit_msg, body, branch_name)
-        elif self.service_type == ServiceType.GITLAB:
-            merge_request = self._gitlab_open_merge_request(commit_msg, body, branch_name)
-        else:
-            raise NotImplementedError
+        try:
+            merge_request = self.repository.create_pr(
+                commit_msg, body, "master", branch_name
+            )
+            merge_request.add_label(labels)
 
-        merge_request.labels = set(labels or [])
-        return merge_request
+        except Exception as exc:
+            raise RuntimeError(f"Failed to create a pull request: {exc}")
```

Comment:
It's not been done in the previous implementation - in general, its a good practice to create project specific exceptions so its clear what errors are raised within the project and to the user.